tests/jni_api.rs

```diff
@@ -508,13 +503,19 @@ pub fn get_auto_long_array_elements_commit() {
     assert_eq!(res[1], 2);
     assert_eq!(res[2], 3);
 
-    auto_ptr.commit();
+    // Change to copy back but don't free (JNI_COMMIT) native buffer
+    auto_ptr.set_release_mode(CopyBackNoFree);
+
+    // Call drop explicitly
+    std::mem::drop(auto_ptr);
 
     // Confirm modification of original Java array
     env.get_long_array_region(java_array, 0, &mut res).unwrap();
     assert_eq!(res[0], 2);
     assert_eq!(res[1], 3);
     assert_eq!(res[2], 4);
+
+    // Native buffer automatically released here anyway
```

Comment:
I might be missing something, but how exactly is it freed? It doesn't seem to get freed on AutoArrays drop, because it is simply stored in a NonNull. If I understand correctly, at this point it is "owned" by the client code.

Reply:
I understand the same. It will be freed when the `AutoArray` object goes out of scope. Not because of the call to `drop()`, but just because of automatic memory management.

Reply:
OK, I've added an extra check to the original `commit` test, and it shows that the original pointer contents change after `drop`, and are therefore no longer valid. 

So, my take here is that what you say is correct: `drop` is releasing the native data associated to the pointer. Not because of the call to Release, but as part of normal (Rust's) memory / borrowing management.

I've also tried explicitly calling `drop` again, on `ptr` and `*ptr`, to see what happens, and couldn't observe any effect.

Finally, I've also tried with the original release mode (`CopyBack`) and couldn't detect any difference in behaviour. This makes me doubt the usefulness of the whole `JNI_COMMIT` thing, in Rust at least.

I'm now thinking that we can probably remove `JNI_COMMIT` enum from `ReleaseMode`. At least, until a clear use case (or mode of use) arises.

Reply:
Interesting, does Rust insert `free` for random pointers inside NonNull 🤔? If it does, why don't we get double-free because of Release(CopyBack) and subsequent "rust magic"?

Reply:
I don't know. It may be `NonNull`, but it also may be the implementation of `ReleaseArrayElements` we're hitting. I guess the only way to know what's happening for sure, would be to get a debugger, and follow the assembler.

Reply:
Another interesting tidbit: I see now that my assertion here in the CI failed, whereas in my local machine, it passed (invalid values).

@dmitry-timofeev, if you guys give me some hints, I'm willing to debug / continue tracking this. Probably will require to compile libjni for debug / etc.

Reply:
Rust definitely shouldn't free anything inside `NonNull`: it doesn't even have a `Drop` implementation. It would be nice to figure out what issue we are hitting and why, but right now I have no idea, either.